### PR TITLE
38 fill missing processor

### PR DESCRIPTION
**Pull request by MarionLeHerisson**
Resolves https://github.com/renanbr/bibtex-parser/issues/38

Added the fillMissingProcessor + tests

### CODE REVIEW

File: src/Processor/FillMissingProcessor.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace RenanBr\BibTexParser\Processor;
+
+class FillMissingProcessor
+{
+    use TagCoverageTrait;
```
Comment (renanbr) [inline src/Processor/FillMissingProcessor.php]: We don't need this trait, as this processor won't change any pre-existing tag.

File: src/Processor/FillMissingProcessor.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+    protected $missingFields;
+
+    public function __construct($missingFields)
+    {
+        $this->missingFields = $missingFields;
+
+        $this->setTagCoverage(['author', 'editor']);
```
Comment (renanbr) [inline src/Processor/FillMissingProcessor.php]: we don't need this

File: src/Processor/FillMissingProcessor.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    public function __invoke(array $entry)
+    {
+        foreach ($this->missingFields as $tag => $value) {
+            if(!array_key_exists($tag, $entry)) {
```
Comment (renanbr) [inline src/Processor/FillMissingProcessor.php]: entry tags are case insensitive, use `TagSearchTrait`, then we can call `tagSearch()` to get the real tag name

File: src/Processor/FillMissingProcessor.php (hunk 1 above)
Comment (MarionLeHerisson) [reply src/Processor/FillMissingProcessor.php]: I wasn't sure about that one, should have asked :smile:

File: src/Processor/FillMissingProcessor.php [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+
+    public function __construct($missingFields)
+    {
+        $this->missingFields = $missingFields;
+    }
+
+    public function __invoke(array $entry)
+    {
+        foreach ($this->missingFields as $tag => $value) {
+            if(!$this->tagSearch($tag, $entry)) {
```
Comment (renanbr) [inline src/Processor/FillMissingProcessor.php]: `tagSearch()` expects as 2nd argument a list of tag names, not the entry itself :wink:

I think storing `array_keys()` result to a variable before the loop solves the case

**Comment by renanbr** [comment]
thank you @MarionLeHerisson

**Pull request merged**